src/omv/imlib/framebuffer.c

```diff
@@ -327,19 +312,21 @@ void framebuffer_flush_buffers(bool fifo_flush) {
 
 int framebuffer_set_buffers(int32_t n_buffers) {
     uint32_t total_size = framebuffer_raw_buffer_size();
-    uint32_t size = total_size / n_buffers;
+    uint32_t vbuff_size = (n_buffers == 1) ?
+                          total_size : FB_ALIGN_SIZE_ROUND_UP(framebuffer->frame_size + sizeof(vbuffer_t));
+    int32_t max_buffers = (n_buffers == 1) ? n_buffers : (total_size / vbuff_size);
+    uint32_t vbuff_count = (n_buffers == -1) ? IM_MIN(max_buffers, 3) : IM_MIN(max_buffers, n_buffers);
 
-    // Error out if frame buffers are smaller than this...
-    if (size < (sizeof(vbuffer_t) + FRAMEBUFFER_ALIGNMENT)) {
+    if (vbuff_count == 0 || vbuff_size < sizeof(vbuffer_t)) {
```

Comment:
```suggestion
    if (vbuff_count == 0 || vbuff_size < (sizeof(vbuffer_t) + FRAMEBUFFER_ALIGNMENT)) {
```

Should leave the alignment padding as part of this.

Reply:
It's already rounded down/up, right ?

Reply:
The vbuffer header yes, but, if the frame buffer storage area itself is 1 byte it will take up 32.

Reply:
I guess the frame size could be 0. So... I suppose it's fine.